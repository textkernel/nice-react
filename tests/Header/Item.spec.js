import React from 'react';
import HeaderMenuItem from '../../lib/Header/Item';

describe('<Header.Menu.Item />', () => {
    it('should render default header menu Item', () => {
        const wrapper = shallow(<HeaderMenuItem
            label="Menu 1"
            context="primary"
            width="125"
            href="http://textkernel.com"
            onClick={ e => {} }
        >
                <HeaderMenuItem label="Item 1" />
                <HeaderMenuItem label="Item 2" />
                <HeaderMenuItem label="Item 3" />
        </HeaderMenuItem>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Header Menu Item with default className/style', () => {
        const wrapper = shallow(
            <HeaderMenuItem className="my-custom-class" style={{ top: 0 }} />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
