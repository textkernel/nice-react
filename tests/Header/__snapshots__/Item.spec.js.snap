// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header.Menu.Item /> should render default header menu Item 1`] = `
<li
  className="highlight primary fixed-width w125"
>
  <a
    className={null}
    href="http://textkernel.com"
    onClick={[Function]}
    target={null}
  >
    Menu 1
  </a>
  <ul>
    <Item
      className={null}
      context={null}
      href="#"
      key=".0"
      label="Item 1"
      level={1}
      onClick={null}
      target={null}
      width={null}
    />
    <Item
      className={null}
      context={null}
      href="#"
      key=".1"
      label="Item 2"
      level={1}
      onClick={null}
      target={null}
      width={null}
    />
    <Item
      className={null}
      context={null}
      href="#"
      key=".2"
      label="Item 3"
      level={1}
      onClick={null}
      target={null}
      width={null}
    />
  </ul>
</li>
`;
